Descriptors and Seeed updates

jadonk opened the pull request:
* Is the Seeed update harmful? Nice to just align if it isn't inaccurate.
* Can you test/use my descriptors update?
* You might want to ignore my CI badge update.

jadonk committed:
Merge pull request #1 from statropy/master

Update to latest

jadonk committed:
Update README.md

jadonk committed:
Update main.yml

jadonk committed:
Update main.yml

Pillar1989 committed:
Support for the latest Beagleconnect hardware. A 24Mhz crystal oscillator is used.

Commit:
Merge branch 'master' of https://github.com/statropy/msp430F55xx_usb_uart_bridge into HEAD

Commit:
Merge branch 'patch-1' of https://github.com/Pillar1989/msp430F55xx_usb_uart_bridge into HEAD

jadonk committed:
Update descriptors.c

jadonk committed:
Update descriptors.h

https://github.com/jadonk/beagleconnect/issues/23
https://github.com/statropy/msp430F55xx_usb_uart_bridge/issues/9

statropy commented on README.md:
```diff
@@ -1,4 +1,4 @@
-![CI](https://github.com/statropy/msp430F55xx_usb_uart_bridge/workflows/CI/badge.svg)
+![CI](https://github.com/jadonk/msp430F55xx_usb_uart_bridge/workflows/CI/badge.svg)
```
The badge here should not change. Seems like the badge for your CI should be in a top-level README with all the submodules listed

statropy commented on .github/workflows/main.yml:
```diff
@@ -32,3 +32,8 @@ jobs:
+      - uses: actions/upload-artifact@v2
+        with:
+          name: usb-uart-bridge
+          path: usb_uart_bridge*.*
```
Can change to `usb_uart_bridge.*` unless you want to capture the build files for the launchpad

statropy commented on hal.c:
```diff
@@ -98,7 +98,7 @@ void hal_init(uint32_t mclkFreq)
 #endif //LAUNCHPAD
     UCS_initClockSignal(
-       UCS_FLLREF,
+       UCS_SMCLK,
```
This should not change

statropy commented on main.c:
```diff
@@ -50,7 +50,7 @@
 #include "ringbuf.h"
 
-#define MCLK_FREQUENCY 20000000
+#define MCLK_FREQUENCY 24000000
```
This should not change. Oscillator speed is independent of system clock.

statropy commented on main.c:
```diff
@@ -89,9 +89,9 @@ int main(void)
     param.selectClockSource = USCI_A_UART_CLOCKSOURCE_SMCLK;
-    param.clockPrescalar = 173;
+    param.clockPrescalar = 208;
     param.firstModReg = 0;
-    param.secondModReg = 5;
+    param.secondModReg = 3;
```
This should not change

statropy committed:
Edit pull-request

Revert system clock changes
Revert CI badge change
Change artifact to only use BeagleConnect build

Signed-off-by: Erik Larson <erik@statropy.com>
